allennlp/tests/common/params_test.py

```diff
@@ -277,3 +277,29 @@ def from_params(cls, params: Params) -> 'A':
                 "a.b.filename": my_file,
                 "a.b.c.c_file": my_other_file
         }
+
+    def test_as_ordered_dict(self):
+        # keyD > keyC > keyE; keyDA > keyDB; Next all other keys alphabetically
+        preference_orders = [["keyD", "keyC", "keyE"], ["keyDA", "keyDB"]]
+        params = Params({"keyC": "valC", "keyB": "valB", "keyA": "valA", "keyE": "valE",
+                         "keyD": {"keyDB": "valDB", "keyDA": "valDA"}})
+        ordered_params_dict = params.as_ordered_dict(preference_orders)
+        ordered_params_dict_str = str(ordered_params_dict)
+        expected_dict_str = str(({'keyD': {'keyDA': 'valDA', 'keyDB': 'valDB'},
+                                  'keyC': 'valC', 'keyE': 'valE',
+                                  'keyA': 'valA', 'keyB': 'valB'}))
+        assert ordered_params_dict_str == expected_dict_str
+
+    def test_to_file(self):
+        # Test to_file works with or without preference orders
+        params_dict = {"keyA": "valA", "keyB": "valB"}
+        ordered_params_dict = {"keyB": "valB", "keyA": "valA"}
```

Comment:
Same here.

Reply:
Sure, let me know if I should explicitly use `OrderedDict` after above comments.

Reply:
done 👍